### ISSUE TITLE
Library options are not exported to calling projects

### ISSUE THREAD

**Issue by pmatalon** [aminya/project_options#275]
I have a header-only library project with numerous options, defined with the cmake functions `option` or `set`, like
````
option(WITH_OPENMP "Enable OpenMP" OFF)
````
The project is packaged with
````
package_project(
  TARGETS my_lib project_options project_warnings libdeps
  INTERFACE_DEPENDENCIES_CONFIGURED ${DEPENDENCIES_CONFIGURED}
  INTERFACE_INCLUDES ${INCLUDE_DIR}
)
````
Now, the CMakeLists.txt of another project has this library as a dependency:
````
project(caller_project CXX)
find_package(my_lib CONFIG REQUIRED)
add_executable(caller_project main.cpp)
target_link_libraries(caller_project INTERFACE my_lib::my_lib my_lib::project_options my_lib::project_warnings my_lib::libdeps)
````

My problem is that I can't set `my_lib`'s options from `caller_project`. I don't see any option related to `my_lib` when I use the `ccmake` command. It seems that the configuration options are not exported to calling projects. 
I tried using `dynamic_project_options` with and without `ENABLE_CACHE`, but no luck...
Thank you for your support.

**Comment by ClausKlein** [comment]
If you want to change the default config values of a sub project set the options before call `FetchContent_MakeAvailable() or `find_package()`! i.e.:
```cmake
include(FetchContent)

FetchContent_Declare(
    glfw
    GIT_REPOSITORY  https://github.com/glfw/glfw.git
    SOURCE_DIR      "${CMAKE_CURRENT_SOURCE_DIR}/glfw-src"
    GIT_TAG         "master"
    GIT_SHALLOW     TRUE
)
set(GLFW_BUILD_WAYLAND  ON)
set(GLFW_BUILD_X11      OFF)
FetchContent_MakeAvailable(glfw)
```

**Comment by pmatalon** [comment]
Thank you for your quick answer. I'll try to clarify my question. 
The options are in my library project, which uses `package_options` for the packaging. I configured the CMakeLists.txt so that If an option is activated, other dependencies are included, compilation options are defined, etc. Now, in a caller project, I want to be able to set those options in my library, thus triggering the linking of other dependencies, compile options for my library, etc. Unfortunately, it seems that my library's options are not exported, so the caller project cannot set them. When I use the `ccmake` command on the caller project, I can see all available options: the options of all dependencies are present, except those of my library.

**Comment by ClausKlein** [comment]
What do you mean with `package_options`? 

They are not at https://github.com/aminya/project_options?tab=readme-ov-file#readme

**Comment by ClausKlein** [comment]
see too https://github.com/aminya/project_options/pull/276/files#r1744556217

**Comment by pmatalon** [comment]
> What do you mean with `package_options`?
> 
> They are not at https://github.com/aminya/project_options?tab=readme-ov-file#readme

Sorry, I meant `project_options`.

**Comment by pmatalon** [comment]
I have updated the text of the issue to include a more detailed `CMakeLists.txt` and re-explain the problem.

**Comment by ClausKlein** [comment]
### Your logic is wrong!

An `interface library` is an` header only` implementation.
The` find_package(OpenMP)` and the resulting **compile_defintions** must be done by the user of your exported `cmake package`!

**Comment by pmatalon** [comment]
I agree that the config can be done by the user, but the work that must be done if an option is activated may be complicated, and it would simplify the user's job if that work was done for him upon a simple option's activation.
Using only standard `cmake` functions, you indeed need a `Config.cmake.in` file to do that. `project_options` does not support it, ok.
